docs/testbed/README.testbed.BgpPeersNumberScaling.md

```diff
@@ -0,0 +1,109 @@
+# Background and Motivation
+
+When deploy a testbed with a great number of virtual ceos neighbors, we will create ceos containers on same server, however, the server doesn't have infinite resources such as memory to deploy that.
+
+To leverage the servers instead of single server, we proposal this design for deploying testbed with multiple servers.
+
+# Table of Contents
+  - [File Content Schema Change](#File-Content-Schema-Change)
+    - [testbed.yml](#testbedyml)
+    - [topo_*.yml](#topo_yml)
+  - [Code Logic Change](#Code-Logic-Change)
+    - [Parameters Constrain Validation](#Parameters-Constrain-Validation)
+    - [Create 802.1Q VLAN](#Create-8021Q-VLAN)
+    - [Create Net and Ceos Containers](#Create-Net-and-Ceos-Containers)
+    - [Create PTF Containers](#Create-PTF-Containers)
+    - [Network Connection](#Network-Connection)
+    - [Automation Script](#Automation-Script)
+    - [Changes To Be Determined](#Changes-To-Be-Determined)
+
+# File Content Schema Change
+
+## testbed.yml
+To make testbeds can be deployed on multiple servers, we deprecated the previous key "server" and add a new key "servers" with a list of servers for deployment.
+
+### Before
+```
+- conf-name: testbed-demo
+  group-name: tb-1
+  topo: t0
+  ptf_image_name: ptf-image-lastest
+  ptf: ptf-tb-1
+  ptf_ip: 1.1.1.1/24
+  ptf_ipv6: fec0::1/64
+  ptf_extra_mgmt_ip: ["2.1.1.1/17"]
+  server: server_1
+  vm_base: VM1000
+  dut:
+    - dut-1
+  inv_name: lab_inv
+```
+
+### After
+When deploy testbed on multiple servers, each server should have one ptf container to provide a environment to run processes like ptf_nn_agent and exbgp. So the ptf container should declare with server.
+
+And because VM hosts in inventory file are children of server and vm_base is related to server, so we declare vm_base with server.
+
+Finally we got a new key "servers" with a list of servers, and each server has its ptf and vm_base declared.
+
+And we add a new key dut_interfaces to indicate which DUT interfaces the server will connect to.
+
+```
+- conf-name: testbed-demo
+  group-name: tb-1
+  topo: t0
+  ptf_image_name: ptf-image-lastest
+  servers:
+    server_1:
+      ptf: ptf-tb-1-1
+      ptf_ip: 1.1.1.1/24
+      ptf_ipv6: fec0::1/64
+      ptf_extra_mgmt_ip: ["2.1.1.1/17"]
+      vm_base: VM1000
+      dut_interfaces: 0-45,49-51  # 0-45 is host_interfaces, 49-51 is bgp peers
+   server_2:
+      ptf: ptf-tb-1-2
+      ptf_ip: 1.1.1.2/24
+      ptf_ipv6: fec0::2/64
+      ptf_extra_mgmt_ip: ["2.1.1.2/17"]
+      vm_base: VM2000
+      dut_interfaces: 47
+   server_3:
+      ptf: ptf-tb-1-3
+      ptf_ip: 1.1.1.3/24
+      ptf_ipv6: fec0::3/64
+      ptf_extra_mgmt_ip: ["2.1.1.3/17"]
+      vm_base: VM3000
+      dut_interfaces: 48
+  dut:
+    - dut-1
+  inv_name: lab_inv
+```
+
+# Code Logic Change
```

Comment:
The detailed code logic change can be reflected in the high level design mentioned above. This will give us a more clear view of everything.